src/DataCollectors/Microsoft.TestPlatform.Extensions.EventLogCollector/EventLogDataCollector.cs

```diff
@@ -149,12 +147,13 @@ internal EventLogDataCollector(IFileHelper fileHelper)
     /// Used by the data collector to send warnings, errors, or other messages
     /// </param>
     /// <param name="dataCollectionEnvironmentContext">Provides contextual information about the agent environment</param>
+    [MemberNotNull(nameof(_events), nameof(_dataSink), nameof(_logger))]
     public override void Initialize(
-        XmlElement configurationElement,
+        XmlElement? configurationElement,
```

Comment:
Functionally we should accept only non-null but that would be a change of behavior so I decided to allow null.